### PR TITLE
ROC-2226: town/city is required in all forms

### PR DESCRIPTION
**Pull request by yarpo**
### JIRA link ###
https://tools.hmcts.net/jira/browse/ROC-2226

### Change description ###
Town / city required

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/app/drafts/models/address.ts [hunk 1]
```diff
@@ -1,8 +1,118 @@
...
+
+    it('should reject when line 1 empty', () => {
+      const errors = validator.validateSync(new Address('', 'line2', 'city', 'postcode'))
+      expect(errors.length).to.equal(1)
+
+      expectValidationError(errors, ValidationErrors.FIRST_LINE_REQUIRED)
+    })
+
+    it('should reject when line 1 too long', () => {
+      const errors = validator.validateSync(new Address(generateString(101), 'line2', 'city', 'postcode'))
```
Comment (rkondratowicz) [inline src/test/app/drafts/models/address.ts]: Could `100` be taken from an existing constant?

File: src/main/app/forms/models/address.ts [hunk 2]
```diff
@@ -26,6 +27,8 @@ export class Address implements Serializable<Address>, CompletableTask {
   line1?: string
   @MaxLength(100, { message: ValidationErrors.SECOND_LINE_TOO_LONG })
   line2?: string
+  @IsDefined({ message: ValidationErrors.CITY_REQUIRED })
+  @IsNotBlank({ message: ValidationErrors.CITY_REQUIRED })
```
Comment (rkondratowicz) [inline src/main/app/forms/models/address.ts]: are both annotations needed?

File: src/test/app/drafts/models/address.ts [hunk 3]
```diff
@@ -1,8 +1,118 @@
...
 
-import { Address } from 'app/forms/models/address'
+import { Address, ValidationErrors } from 'app/forms/models/address'
+import { Address as ClaimAddress } from 'claims/models/address'
+import { Validator } from 'class-validator'
+import { expectValidationError } from '../../forms/models/validationUtils'
+import * as randomstring from 'randomstring'
 
 describe('Address', () => {
+  describe('validation', () => {
```
Comment (geek-soft) [inline src/test/app/drafts/models/address.ts]: I do not think this is appropriate place for validation tests. We already have validation tests in `src/test/app/forms/models/address.ts` and these tests should have been amended.

File: src/test/app/forms/models/address.ts [hunk 4]
```diff
@@ -2,20 +2,38 @@
...
+import { Address as ClaimAddress } from 'claims/models/address'
 
 describe('Address/CorrespondenceAddress', () => {
   [
-    { classFunction: Address, validationErrors: AddressValidationErrors },
-    { classFunction: CorrespondenceAddress, validationErrors: CorrespondenceAddressValidationErrors }
+    {
+      classFunction: Address,
+      validationErrors: AddressValidationErrors,
+      validationConstants: CorrespondenceAddressValidationConstants
```
Comment (geek-soft) [inline src/test/app/forms/models/address.ts]: It does not really matter from runtime perspective, but it should be `AddressValidationConstants` here and `CorrespondenceAddressValidationConstants` at line 30.

File: src/test/app/forms/models/address.ts [hunk 5]
```diff
@@ -2,20 +2,38 @@
...
+    },
+    {
+      classFunction: CorrespondenceAddress,
+      validationErrors: CorrespondenceAddressValidationErrors,
+      validationConstants: AddressValidationConstants
+    }
   ].forEach((testInput) => {
     const ClassFunction = testInput.classFunction
     const ValidationErrors = testInput.validationErrors
+    const tooLongAddress: number = testInput.validationConstants.ADDRESS_MAX_LENGTH + 1
```
Comment (geek-soft) [inline src/test/app/forms/models/address.ts]: These names are confusing to me. When I saw `generateString(tooLongAddress)` and I did not remember the method signature I couldn't know what the value behind `tooLongAddress` might be. For me if we had `addressMaxLength + 1` is quite descriptive unless we can came with good name for `addressMaxLength + 1` like `exceededAddressLength`

File: src/test/app/forms/models/address.ts (hunk 4 above)
Comment (yarpo) [reply src/test/app/forms/models/address.ts]: fixed

File: src/test/app/forms/models/address.ts (hunk 5 above)
Comment (yarpo) [reply src/test/app/forms/models/address.ts]: done

**Comment by yarpo** [comment]
I don't know why I cannot merge it.

Tests has passed: https://dev-build.reform.hmcts.net/blue/organizations/jenkins/hmcts%2Fcmc-citizen-frontend/detail/PR-33/4/pipeline/

**Pull request merged**